src/gt4py/backend/templates/computation.src.in

```diff
@@ -72,38 +73,61 @@ using backend_t = gt::backend::{{ gt_backend }};
 static constexpr gt::uint_t level_offset_limit = {{ k_axis.offset_limit }};
 
 using axis_t =
-    gridtools::axis<{{ k_axis.n_intervals }}, /* NIntervals */
-                    gt::axis_config::offset_limit<level_offset_limit>>;
+    gt::axis<{{ k_axis.n_intervals }}, /* NIntervals */
+             gt::axis_config::offset_limit<level_offset_limit>>;
 
 // These halo sizes are used to determine the sizes of the temporaries
 static constexpr gt::uint_t halo_size_i = {{ halo_sizes[0] }};
 static constexpr gt::uint_t halo_size_j = {{ halo_sizes[1] }};
 static constexpr gt::uint_t halo_size_k = {{ halo_sizes[2] }};
 
 // Storage definitions
+{%- for axes in used_axes %}
 template <int Id>
+{%- if axes.name == "ijk" %}
 using storage_info_t =
     gt::storage_traits<backend_t>::storage_info_t<
         Id, 3,
+{%- else %}
+using storage_info_{{ axes.name }}_t =
+    gt::storage_traits<backend_t>::special_storage_info_t<
+        Id, gt::selector<{{ axes.selector }}>,
+{%- endif %}
         gt::halo<0, 0, 0 /* not used */>>;
+{%- endfor %}
 
+{%- for axes in used_axes %}
 template<typename T, int Id>
+{%- if axes.name == "ijk" %}
 using data_store_t =
     gt::storage_traits<backend_t>::data_store_t<T, storage_info_t<Id>>;
+{%- else %}
+using data_store_{{ axes.name }}_t =
+    gt::storage_traits<backend_t>::data_store_t<T, storage_info_{{ axes.name }}_t<Id>>;
+{%- endif %}
+{%- endfor %}
 
 // Placeholder definitions
 {% set arg_counter = 0 -%}
 {%- for field in arg_fields %}
+{%- if field.axes == "ijk" %}
 using p_{{ field.name }} = gt::arg<{{ arg_counter + loop.index0 }}, data_store_t<{{ field.dtype }}, {{ field.layout_id }}>>;
+{%- else %}
+using p_{{ field.name }} = gt::arg<{{ arg_counter + loop.index0 }}, data_store_{{ field.axes }}_t<{{ field.dtype }}, {{ field.layout_id }}>>;
+{%- endif %}
 {%- endfor %}
 {% set arg_counter = arg_counter + arg_fields|length %}
 {% for param in parameters %}
-using p_{{ param.name }} = gt::arg<{{ arg_counter + loop.index0 }}, gt::global_parameter<{{ param.dtype }}>>;
+using p_{{ param.name }} = gt::arg<{{ arg_counter + loop.index0 }}, gt::global_parameter<backend_t, {{ param.dtype }}>>;
```

Comment:
Is `backend_t` is needed here? You may have gotten this at one point from one of my PRs, but I'm not sure it's really needed after all.